angular/src/app/services/wallet-manager.ts

```diff
@@ -186,14 +189,58 @@ export class WalletManager {
     return data;
   }
 
+  async selectUtxos(utxos: { txId: string; vout: number; value: number }[], targets: { address: string; value: number }[], feeRate: number): Promise<CoinSelectionResult> {
+    // When user manually edits fee, the value becomes string and not number and the coinselect library is very strict on type.
+    if (typeof String(feeRate)) {
+      feeRate = Number(feeRate);
+    }
+
+    // The coin select library will result in fee rate that is right below what is specified, so for example
+    // 10 can result in fees at around 9.9-9.5 etc. and that won't be accepted by the nodes when broadcasted.
```

Comment:
this is wired is that the lib logic? so coin selection is delegated to a lower level lib? this is fine it means the logic was probably tested quite extensively

Reply:
Yes, it's used by multiple wallets already afaik.